StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -293,6 +300,68 @@ extension EmbeddedPaymentElement: EmbeddedFormViewControllerDelegate {
 }
 
 extension EmbeddedPaymentElement {
+    
+    func _confirm() async -> (result: PaymentSheetResult, deferredIntentConfirmationType: STPAnalyticsClient.DeferredIntentConfirmationType?) {
+        let result: (PaymentSheetResult, STPAnalyticsClient.DeferredIntentConfirmationType?) = await {
+            
+            let authContext: STPAuthenticationContext? = {
+                switch configuration.formSheetAction {
+                case .confirm:
+                    if let formViewController {
+                        return formViewController
+                    }
+                    if let presentingViewController {
+                        return STPAuthenticationContextWrapper(presentingViewController: presentingViewController)
+                    }
+                    return nil
+                case .continue:
+                    // formViewController is never currently presented during confirmation in continue mode
+                    if let presentingViewController {
+                        return STPAuthenticationContextWrapper(presentingViewController: presentingViewController)
+                    }
+                    return nil
+                }
+            }()
+            
+            guard let authContext else {
+                return (.failed(error: PaymentSheetError.unknown(debugDescription: "Unexpectedly found nil on both bottomSheetController and presentingViewController.")), STPAnalyticsClient.DeferredIntentConfirmationType.none)
+            }
+            
+            guard let paymentOption = _paymentOption else {
+                return (.failed(error: PaymentSheetError.unknown(debugDescription: "Unexpectedly found nil payment option.")),
+                        STPAnalyticsClient.DeferredIntentConfirmationType.none)
+            }
+            
+            return await PaymentSheet.confirm(configuration: configuration,
+                                                    authenticationContext: authContext,
+                                                    intent: intent,
+                                                    elementsSession: elementsSession,
+                                                    paymentOption: paymentOption,
+                                                    paymentHandler: paymentHandler,
+                                                    integrationShape: .embedded)
+        }()
+        
+        // Wait for the last update to finish and fail if didn't succeed. A failure means the view is out of sync with the intent and could e.g. not be showing a required mandate.
+        if let latestUpdateTask {
+            switch await latestUpdateTask.value {
```

Comment:
I'm a bit confused on how this function works. It seems like you'd want to wait for latestUpdateTask to finish before attempting to confirm, but the way the code is written, it looks like `await PaymentSheet.confirm` could execute before checking `latestUpdateTask`?

Reply:
`result` is a closure that gets [invoked](https://github.com/stripe/stripe-ios/pull/4242/files#diff-b66bb3ee17a6e8b6b2c784056b390b3ee60b34d788c38ba8aac66ef1ded1558dR349) if the last update has completed or just [returns](https://github.com/stripe/stripe-ios/pull/4242/files#diff-b66bb3ee17a6e8b6b2c784056b390b3ee60b34d788c38ba8aac66ef1ded1558dR362) result if there was no last update. We have a test for this [here](https://github.com/stripe/stripe-ios/pull/4242/files#diff-8908c6f71c82abec6b4f4989d9fa0ce5750cfa6e76702fb67d8c27ae86e9e1d7R175).

Reply:
You're understanding is right, mine wasn't. Fixed this.